File: server/lobby/payload.go
package lobby

// CreateLobbyPayload represents the payload for creating a new lobby
type CreateLobbyPayload struct {
	GameName   string `json:"gameName"`
	MaxPlayers int    `json:"maxPlayers"`
}

// CreateLobbyResponse represents the response after creating a lobby
type CreateLobbyResponse struct {
	LobbyID    string `json:"lobbyId"`
	GameName   string `json:"gameName"`
	MaxPlayers int    `json:"maxPlayers"`
	IsHost     bool   `json:"isHost"`
}

// JoinLobbyPayload represents the payload for joining a lobby
type JoinLobbyPayload struct {
	LobbyID string `json:"lobbyId"`
}

// PlayerJoinedPayload represents the payload when a player joins a lobby
type PlayerJoinedPayload struct {
	UserID   string `json:"userId"`
	Username string `json:"username"`
	AvatarID string `json:"avatarId"`
	IsHost   bool   `json:"isHost"`
	Ready    bool   `json:"ready"`
}

// PlayerLeftPayload represents the payload when a player leaves a lobby
type PlayerLeftPayload struct {
	UserID string `json:"userId"`
}

// HostChangedPayload represents the payload when the host changes
type HostChangedPayload struct {
	NewHostID string `json:"newHostId"`
}

// ReadyChangedPayload represents the payload when a player's ready status changes
type ReadyChangedPayload struct {
	UserID string `json:"userId"`
	Ready  bool   `json:"ready"`
}

// GameStartingPayload represents the payload when a game is starting
type GameStartingPayload struct {
	LobbyID string `json:"lobbyId"`
	GameID  string `json:"gameId"`
}

// ReorderPlayersPayload represents the payload for reordering players
type ReorderPlayersPayload struct {
	PlayerOrder []string `json:"playerOrder"`
}

// ErrorPayload represents an error message payload
type ErrorPayload struct {
	Message string `json:"message"`
}

// LobbyStatePayload represents the complete state of a lobby
type LobbyStatePayload struct {
	LobbyID    string       `json:"lobbyId"`
	GameName   string       `json:"gameName"`
	MaxPlayers int          `json:"maxPlayers"`
	HostID     string       `json:"hostId"`
	Players    []PlayerInfo `json:"players"`
}

// PlayerInfo represents information about a player in the lobby
type PlayerInfo struct {
	ID       string `json:"id"`
	Username string `json:"username"`
	AvatarID string `json:"avatarId"`
	Ready    bool   `json:"ready"`
	IsHost   bool   `json:"isHost"`
}

File: app/services/payload.ts
// Message types
export type MessageType =
  | 'create_lobby'
  | 'join_lobby'
  | 'leave_lobby'
  | 'toggle_ready'
  | 'start_game'
  | 'reorder_players'
  | 'lobby_created'
  | 'player_joined'
  | 'player_left'
  | 'player_ready_changed'
  | 'game_starting'
  | 'players_reordered'
  | 'host_changed'
  | 'lobby_state'
  | 'lobby_closed'
  | 'list_lobbies'
  | 'error';

// Payload types matching Go backend
export interface CreateLobbyPayload {
  gameName: string;
  maxPlayers: number;
}

export interface CreateLobbyResponse {
  lobbyId: string;
  gameName: string;
  maxPlayers: number;
  isHost: boolean;
}

export interface JoinLobbyPayload {
  lobbyId: string;
}

export interface PlayerJoinedPayload {
  userId: string;
  username: string;
  avatarId: string;
  isHost: boolean;
  ready: boolean;
}

export interface PlayerLeftPayload {
  userId: string;
}

export interface HostChangedPayload {
  newHostId: string;
}

export interface ReadyChangedPayload {
  userId: string;
  ready: boolean;
}

export interface GameStartingPayload {
  lobbyId: string;
  gameId: string;
}

export interface ReorderPlayersPayload {
  playerOrder: string[];
}

export interface ErrorPayload {
  message: string;
}

export interface PlayerInfo {
  id: string;
  username: string;
  avatarId: string;
  ready: boolean;
  isHost: boolean;
}

export interface LobbyState {
  lobbyId: string;
  gameName: string;
  maxPlayers: number;
  hostId: string;
  players: PlayerInfo[];
}
